create.ps1

```diff
@@ -303,127 +292,126 @@ try {
-                    }
-                    else {
-                        throw $_                            
-                    }
+
+                if (-Not($actionContext.DryRun -eq $true)) {
+                    $responseCreateUser = Invoke-RestMethod @splatCreateUserParams -Verbose:$false
+                }
+                else {
+                    Write-Warning "Will send: $($splatCreateUserParams.Body)"
                 }
-            } while ($uniqueness -ne $true -and $Iterator -lt $maxIterations)
-            
-            if (!($uniqueness -eq $true)) {
-                throw "A user with the 'ysisInitials' '$($account.ysisInitials)' already exists. YSIS-Initials out of iterations!"                      
-            }            
-            if ($actionContext.DryRun -eq $true) {
-            
-                Write-Warning "Account with username [$($account.UserName)] and discipline [$($mappedObject.Discipline)] will be created."
+                $uniqueness = $true
+
+                $outputContext.AccountReference = $responseCreateUser.id
+
                 $outputContext.AuditLogs.Add([PSCustomObject]@{
-                        Action  = "CreateAccount" # Optionally specify a different action for this audit log
-                        Message = "Account with username [$($account.UserName)] and discipline [$($mappedObject.Discipline)] will be created."
+                        Action  = "CreateAccount"
+                        Message = "Created account with username $($account.UserName)"
                         IsError = $false
                     })
+
+            }
+            catch {
+                $ex = $PSItem
+                $errorObj = Resolve-YsisError -ErrorObject $ex
+                Write-Warning "$Iterator : $($_.Exception.Response.StatusCode) - $($errorObj.FriendlyMessage)"
+
+                if ($_.Exception.Response.StatusCode -eq 'Conflict' -and $($errorObj.FriendlyMessage) -match "A user with the 'ysisInitials'") {
+                    $Iterator++
+                    Write-Warning "Ysis-Initials in use, trying with [$($account.'urn:ietf:params:scim:schemas:extension:ysis:2.0:User'.ysisInitials)]"
+                }
+                else {
+                    throw $_
+                }
             }
+        } while ($uniqueness -ne $true -and $Iterator -lt $maxIterations)
 
-            # add ID to export data
-            $account | Add-Member -MemberType NoteProperty -Name id -Value $responseCreateUser.id -Force
-            # $outputContext.Data = $account 
+        if (-not($uniqueness -eq $true)) {
+            throw "A user with the 'ysisInitials' '$($account.'urn:ietf:params:scim:schemas:extension:ysis:2.0:User'.ysisInitials)' already exists. YSIS-Initials out of iterations"
         }
-    }  
+
+        if ($actionContext.DryRun -eq $true) {
+
+            Write-Warning "[DryRun] Account with username [$($account.UserName)] and discipline [$($mappedObject.Discipline)] will be created."
```

Comment:
Please check al scripts for DryRun. Sometimes write-warning is used, sometimes write-information and sometimes only auditlogs